src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/instances/arm/matmul.cpp

```diff
@@ -0,0 +1,142 @@
+                                                             ::testing::Values(ElementType::undefined),
+                                                             ::testing::Values(ElementType::undefined),
+                                                             ::testing::Values(utils::InputLayerType::CONSTANT),
+                                                             ::testing::Values(ov::test::utils::DEVICE_CPU),
+                                                             ::testing::Values(emptyAdditionalConfig()));
+std::vector<fusingSpecificParams> fusingParamsSet3D_smoke_f16 {
+        emptyFusingSpec,
+        fusingBias,
+        fusingRelu
+};
+const auto fullyConnectedParams3D_smoke_f16 = ::testing::Combine(::testing::ValuesIn(IS3D_smoke()),
+                                                             ::testing::Values(ElementType::f16),
+                                                             ::testing::Values(ElementType::undefined),
+                                                             ::testing::Values(ElementType::undefined),
+                                                             ::testing::Values(utils::InputLayerType::CONSTANT),
+                                                             ::testing::Values(ov::test::utils::DEVICE_CPU),
+                                                                 ::testing::Values(
+                                                                         ov::AnyMap({ov::hint::inference_precision(ov::element::f16)})));
+const auto testParams3D_smoke = ::testing::Combine(fullyConnectedParams3D_smoke,
+                                                   ::testing::Values(MatMulNodeType::FullyConnected),
+                                                   ::testing::ValuesIn(fusingParamsSet3D_smoke),
+                                                   ::testing::ValuesIn(filterCPUInfo(filterSpecificParams())));
+const auto testParams3D_smoke_f16 = ::testing::Combine(fullyConnectedParams3D_smoke_f16,
+                                                   ::testing::Values(MatMulNodeType::FullyConnected),
+                                                   ::testing::ValuesIn(fusingParamsSet3D_smoke_f16),
+                                                   ::testing::ValuesIn(filterCPUInfo(filterSpecificParams())));
+INSTANTIATE_TEST_SUITE_P(smoke_FC_3D, MatMulLayerCPUTest, testParams3D_smoke, MatMulLayerCPUTest::getTestCaseName);
+INSTANTIATE_TEST_SUITE_P(smoke_FC_3D_f16, MatMulLayerCPUTest, testParams3D_smoke_f16, MatMulLayerCPUTest::getTestCaseName);
+
+const std::vector<ShapeRelatedParams> IS = {
+        {static_shapes_to_test_representation({{1, 2, 32, 120}, {120, 5}}), {false, false}},
+        {static_shapes_to_test_representation({{1, 2, 32, 120}, {120, 5}}), {true, false}},
+        {static_shapes_to_test_representation({{1, 2, 32, 120}, {120, 5}}), {false, true}},
+        {static_shapes_to_test_representation({{1, 2, 32, 120}, {120, 5}}), {true, true}},
```

Comment:
We need to actually complete the tests refactoring for FullyConnected node (to add common tests, etc). Let's do it in scope of a follow up PR.

Reply:
@EgorDuplensky created issue CVS-145273